### ISSUE TITLE
same fairness metric with different `by` arguments are merged in `autoplot()`

### ISSUE THREAD

**Issue by simonpcouch** [tidymodels/tune#773]
For the most part, `autoplot()` methods "just work" with fairness metrics, but we do see a hiccup when the same fairness metric is applied to two different columns. e.g.:

``` r
library(tune)

mtcars$vs <- as.factor(mtcars$vs) 
mtcars$cyl <- as.factor(mtcars$cyl)
mtcars$am <- as.factor(mtcars$am)

set.seed(1)
res <- tune_grid(
  parsnip::nearest_neighbor("classification", "kknn", neighbors = tune()), 
  vs ~ mpg + hp, 
  resamples = rsample::bootstraps(mtcars, 3), 
  grid = 3,
  metrics =
    yardstick::metric_set(
      yardstick::demographic_parity(cyl),
      yardstick::demographic_parity(am)
    )
)

autoplot(res)
```

![](https://i.imgur.com/8eikReB.png)

<sup>Created on 2023-12-01 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by simonpcouch** [comment]
Feels worth mentioning that this is also the case when the metric factory's output has been assigned to something, i.e.:

``` r
library(tune)
library(yardstick)

mtcars$vs <- as.factor(mtcars$vs) 
mtcars$cyl <- as.factor(mtcars$cyl)
mtcars$am <- as.factor(mtcars$am)

set.seed(1)

demographic_parity_by_cyl <- demographic_parity(cyl)
demographic_parity_by_am <- demographic_parity(am)

res <- tune_grid(
  parsnip::nearest_neighbor("classification", "kknn", neighbors = tune()), 
  vs ~ mpg + hp, 
  resamples = rsample::bootstraps(mtcars, 3), 
  grid = 3,
  metrics =
    yardstick::metric_set(
      demographic_parity_by_cyl,
      demographic_parity_by_am
    )
)
```

**Comment by simonpcouch** [comment]
Before #790:

``` r
library(tidymodels)

mtcars$vs <- as.factor(mtcars$vs) 
mtcars$cyl <- as.factor(mtcars$cyl)
mtcars$am <- as.factor(mtcars$am)

set.seed(1)

# plot_regular_grid() ----------------------------------------------------------
res <- tune_grid(
  nearest_neighbor("classification", "kknn", neighbors = tune()), 
  vs ~ mpg + hp, 
  resamples = bootstraps(mtcars, 3), 
  grid = 3,
  metrics =
    yardstick::metric_set(
      demographic_parity(cyl),
      demographic_parity(am)
    )
)

autoplot(res)
```

![](https://i.imgur.com/sld94zM.png)

``` r

# plot_marginals() -------------------------------------------------------------
res <- tune_grid(
  nearest_neighbor("classification", "kknn", neighbors = tune(), weight_func = tune()), 
  vs ~ mpg + hp, 
  resamples = bootstraps(mtcars, 3), 
  metrics =
    yardstick::metric_set(
      demographic_parity(cyl),
      demographic_parity(am)
    )
)

autoplot(res)
```

![](https://i.imgur.com/T8ZYeNk.png)

``` r

# iterative searches -----------------------------------------------------------
res <- tune_bayes(
  nearest_neighbor("classification", "kknn", neighbors = tune()), 
  vs ~ mpg + hp, 
  resamples = bootstraps(mtcars, 3), 
  metrics =
    yardstick::metric_set(
      roc_auc,
      demographic_parity(cyl),
      demographic_parity(am)
    )
)
#> ! No improvement for 10 iterations; returning current results.

#   plot_param_vs_iter() -------------------------------------------------------
#   needs no changes - not a parameter

#   plot_perf_vs_iter() --------------------------------------------------------
autoplot(res, type = "performance")
```

![](https://i.imgur.com/PldZID3.png)

<sup>Created on 2023-12-18 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by simonpcouch** [comment]
Closed in #790. :)
